pkg/otlp/metrics/ttlcache_test.go

```diff
@@ -28,29 +29,57 @@ func newTestCache() *ttlCache {
 var dims = &Dimensions{name: "test"}
 
 func TestMonotonicDiffUnknownStart(t *testing.T) {
-	startTs := uint64(0) // equivalent to start being unset
-	prevPts := newTestCache()
-
-	_, firstPoint, dropPoint := prevPts.MonotonicDiff(dims, startTs, 1, 5)
-	assert.True(t, firstPoint)
-	assert.False(t, dropPoint, "first point")
-
-	_, firstPoint, dropPoint = prevPts.MonotonicDiff(dims, startTs, 1, 6)
-	assert.False(t, firstPoint)
-	assert.True(t, dropPoint, "new ts == old ts")
-
-	_, firstPoint, dropPoint = prevPts.MonotonicDiff(dims, startTs, 0, 0)
-	assert.False(t, firstPoint)
-	assert.True(t, dropPoint, "new ts < old ts")
-
-	_, firstPoint, dropPoint = prevPts.MonotonicDiff(dims, startTs, 2, 2)
-	assert.True(t, firstPoint, "new < old => there has been a reset")
-	assert.False(t, dropPoint, "first point")
-
-	dx, firstPoint, dropPoint := prevPts.MonotonicDiff(dims, startTs, 3, 4)
-	assert.False(t, firstPoint)
-	assert.False(t, dropPoint)
-	assert.Equal(t, 2.0, dx, "expected diff 2.0 with (0,2,2) value")
+	t.Run("diff", func(t *testing.T) {
+		startTs := uint64(0) // equivalent to start being unset
+		prevPts := newTestCache()
+
+		_, firstPoint, dropPoint := prevPts.MonotonicDiff(dims, startTs, 1, 5)
+		assert.True(t, firstPoint)
+		assert.False(t, dropPoint, "first point")
+
+		_, firstPoint, dropPoint = prevPts.MonotonicDiff(dims, startTs, 1, 6)
+		assert.False(t, firstPoint)
+		assert.True(t, dropPoint, "new ts == old ts")
+
+		_, firstPoint, dropPoint = prevPts.MonotonicDiff(dims, startTs, 0, 0)
+		assert.False(t, firstPoint)
+		assert.True(t, dropPoint, "new ts < old ts")
+
+		_, firstPoint, dropPoint = prevPts.MonotonicDiff(dims, startTs, 2, 2)
+		assert.True(t, firstPoint, "new < old => there has been a reset")
+		assert.False(t, dropPoint, "first point")
+
+		dx, firstPoint, dropPoint := prevPts.MonotonicDiff(dims, startTs, 3, 4)
+		assert.False(t, firstPoint)
+		assert.False(t, dropPoint)
+		assert.Equal(t, 2.0, dx, "expected diff 2.0 with (0,2,2) value")
```

Comment:
I realize this was 'wrong' before your PR, but since you are adding new tests I am going to suggest we rewrite this as something like the following:

```go
// May not compile as-is, just wrote this without checking

points  := []struct{
  ts uint64
  val float64
  expectFirstPoint bool
  expectDropPoint bool
  dropPointMessage string
}{
// ...
}

var dx float64
startTs := uint64(0) // equivalent to start being unset
prevPts := newTestCache()
for _, point := range points {
  dx, firstPoint, dropPoint := prevPts.MonotonicDiff(dims, startTs, point.ts, point,val)
  assert.Equal(t, point.expectFirstPoint, firstPoint)
  assert.Equal(t, point.expectDropPoint, dropPoint, point.dropPointMessage)
}

assert.Equal(t, 2.0, dx, "...")
```

Reply:
Done. I also tried to do the same with `TestMonotonicDiffKnownStart`, but needed to split into multiple slices for the resets.

Reply:
Awesome, looks much better IMO, thanks :)